lib/api/src/sys/instance.rs

```diff
@@ -129,6 +130,14 @@ impl Instance {
             })
             .collect::<Exports>();
 
+        // If the memory is imported then also export it for backwards compatibility reasons
+        // (many will assume the memory is always exported) - later we can remove this
+        if exports.get_memory("memory").is_err() {
+            if let Some(memory) = externs.iter().find(|a| a.ty(store).memory().is_some()) {
+                exports.insert("memory", memory.clone());
+            }
+        }
```

Comment:
I don't think this is right

Reply:
without exporting it (for things compiled without `atomics`) we would need a way of passing the memory to WASIX in some other construct

Reply:
we could always try removing this and seeing if the regression fails - I'm not 100% sure its needed looking at the code again - I seem to remember regression issues when I tried to take it out (hence the comment)

Reply:
stays for now